### PR TITLE
Added custom validation for tag/sensor

### PR DESCRIPTION
**Pull request by bhasin85**
#373 

@julian1  @xhoenner

### CODE REVIEW

**Comment by julian1** [comment]
@bhasin85 Are the created objects now all saved in the one transaction?

File: grails-app/domain/au/org/emii/aatams/Sensor.groovy [hunk 1]
```diff
@@ -92,6 +94,23 @@ class Sensor implements Embargoable {
         return null
     }
 
+    static Map<String, List<String>> validTransmitterMap = new HashMap<String, List<String>>() {{
+        def transmitterTypes1 = Arrays.asList("PINGER", "RANGE TEST");
```
Comment (julian1) [inline grails-app/domain/au/org/emii/aatams/Sensor.groovy]: Should these values be populated from the database rather than be hard-coded?

Comment (jonescc) [inline grails-app/domain/au/org/emii/aatams/Sensor.groovy]: Strongly agree that these database values should not be hardcoded in the code if possible. Should be externally configured in config or database.

**Comment by bhasin85** [comment]
@julian1 Objects are getting created in one transaction. Ready for review.

File: grails-app/domain/au/org/emii/aatams/CodeMap.groovy [hunk 2]
```diff
@@ -7,7 +7,7 @@ package au.org.emii.aatams
  */
 class CodeMap  {
```
Comment (julian1) [inline grails-app/domain/au/org/emii/aatams/CodeMap.groovy]: This looks like straight Java, should it have a java file extension? I am surprised the grails compiler is able to handle it.

**Comment by julian1** [comment]
Looks really good.

**Pull request merged**